standard/expressions.md

```diff
@@ -1150,6 +1151,200 @@ object o = (new int[3])[1];
 
 A *primary_expression* that consists of a *literal* ([§7.4.5](lexical-structure.md#745-literals)) is classified as a value.
 
+### §interpolated-string-expressions Interpolated string expressions
+
+An *interpolated_string_expression* consists of a `$` character immediately followed by text within `"` characters. Within the quoted text there are zero or more **_interpolations_** delimited by `{` and `}` characters, each of which encloses an *expression* and optional formatting specifications.
+
+Interpolated string expressions have two forms; regular (*interpolated_regular_string_expression*)
+and verbatim (*interpolated_verbatim_string_expression*); which are lexically similar to, but differ semantically from, the two forms of string
+literals ([§7.4.5.6](lexical-structure.md#7456-string-literals)).
+
+```ANTLR
+interpolated_string_expression
+    : interpolated_regular_string_expression
+    | interpolated_verbatim_string_expression
+    ;
+
+// interpolated regular string expressions
+
+interpolated_regular_string_expression
+    : Interpolated_Regular_String_Start Interpolated_Regular_String_Mid?
+      ('{' regular_interpolation '}' Interpolated_Regular_String_Mid?)* Interpolated_Regular_String_End
+    ;
+
+regular_interpolation
+    : expression (',' interpolation_minimum_width)? Regular_Interpolation_Format?
+    ;
+
+interpolation_minimum_width
+    : constant_expression
+    ;
+
+Interpolated_Regular_String_Start
```

Comment:
I can't remember the different casing rules we've got, and I probably couldn't check whether these are appropriate anyway. I strongly suspect they're all fine, but I bring it up so that someone with more of a clue can review that aspect specifically.

Reply:
I suspect it looks a bit off as this is the first place where we really intermingle lexer & parser rules a lot. The casing is correct or ANTLR would have barfed.